packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -153,6 +156,69 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
     };
   }
 
+  /**
+   * Bootstrap Action
+   */
+  public async bootstrap(cx: ICloudAssemblySource, userEnvironmentSpecs: string[], options: BootstrapOptions = {}): Promise<void> {
```

Comment:
This is the biggest open discussion: 

`cx` should not be a required option. It's really that the user needs to provide either a list of `environments` or a `cx. Typically we solve this with a helper class:

```
BootstrapEnvironments.fromList();
BootstrapEnvironments.fromCloudAssemblySource(cx);
```

Think about what the class would look like.

Reply:
See the new function signature:
```
  public async bootstrap(options: BootstrapOptions & {
    environments?: string[];
    cloudAssembly?: ICloudAssemblySource;
  }): Promise<void> {
```